### PR TITLE
Avoid mutating the cached deployment config in image change controller

### PR DESCRIPTION
**Pull request by smarterclayton**
Will ensure we handle retries correctly.

[test]

### CODE REVIEW

**Comment by smarterclayton** [comment]
@kargakis

**Comment by kargakis** [comment]
Bump [retries](https://github.com/openshift/origin/blob/010d8ad15b705c668c5178b8ffb8100a5cd62244/pkg/deploy/controller/imagechange/factory.go#L72) to 2 or 3?

File: pkg/deploy/controller/imagechange/controller.go [hunk 1]
```diff
@@ -88,6 +90,19 @@ func (c *ImageChangeController) Handle(stream *imageapi.ImageStream) error {
 				if !names.Has(container.Name) {
 					continue
 				}
+
+				if !hasImageChange {
+					// create a copy prior to mutation
+					result, err := kapi.Scheme.Copy(config)
```
Comment (kargakis) [inline pkg/deploy/controller/imagechange/controller.go]: deployutil.DeploymentConfigDeepCopy

Comment (smarterclayton) [inline pkg/deploy/controller/imagechange/controller.go]: Why do we have a special deep copy?

Comment (smarterclayton) [inline pkg/deploy/controller/imagechange/controller.go]: I guess it's fine.

**Comment by smarterclayton** [comment]
Will bump retries and use helper.

**Comment by smarterclayton** [comment]
updated

**Comment by kargakis** [comment]
LGTM [merge]

**Comment by smarterclayton** [comment]
Fixed in the other PR.

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
Hrm... or not.

**Comment by smarterclayton** [comment]
Was a coding error, didn't properly copy the params from the new type.

**Comment by smarterclayton** [comment]
Third time's the charm.

**Comment by smarterclayton** [comment]
[test] [merge]

**Pull request merged**